src/Bicep.Core/Emit/TemplateWriter.cs

```diff
@@ -211,23 +211,41 @@ private ObjectSyntax AddDecoratorsToBody(StatementSyntax statement, ObjectSyntax
         private void EmitParameter(JsonTextWriter jsonWriter, ParameterSymbol parameterSymbol, ExpressionEmitter emitter)
         {
             var declaringParameter = parameterSymbol.DeclaringParameter;
-            if (SyntaxHelper.TryGetPrimitiveType(declaringParameter) is not TypeSymbol primitiveType)
+
+            var properties = new List<ObjectPropertySyntax>();
+            if (parameterSymbol.Type is ResourceType resourceType)
```

Comment:
> One pattern I can think of is adding one or more properties on the symbol that is/are computed off of syntax. That would make it more obvious that the emit logic is directed by syntax rather than a type.

This works for now, but won't work once we extensibility support for this feature, because non-syntactic context will mattter.

For example we will need to include information about the provider, which is not driven by the (local) syntax of the parameter output. I will improve this for now based on your suggestion and we can cross that bridge later.